m2cgen/assemblers/boosting.py

```diff
@@ -134,7 +134,7 @@ def _assemble_tree(self, tree):
         if "leaf" in tree:
             return ast.NumVal(tree["leaf"])
 
-        threshold = ast.NumVal(tree["split_condition"])
+        threshold = ast.NumVal(np.float32(tree["split_condition"]))
```

Comment:
Maybe a more general solution will be to add an optional `dtype` constructor argument? I mean,
```
class NumVal(NumExpr):
    def __init__(self, value, dtype=np.float64):
```

Reply:
That's a good idea 👍